src/services/MetricsService/MetricsService.ts

```diff
@@ -72,32 +88,63 @@ export class DefaultMetricsService implements MetricsService {
         message: 'publishing new metrics',
         payload: metrics,
       });
-      return this.metricsPublisher.publish(metrics).then(() => {
-        if (forcePush) {
-          return this.triggerPush();
-        }
-      });
+      return this.metricsPublisher
+        .publish(metrics)
+        .then(() => {
+          if (forcePush) {
+            return this.triggerPush();
+          }
+        })
+        .then(() => {});
     });
   }
 
-  sendMetrics(): Promise<void> {
-    return this.serialPromiseQueue.add(() => this.triggerPush());
+  sendDailyMetrics(): Promise<void> {
+    const pushAndMarkLastUploadedDateTime = (): Promise<void> => {
+      return this.triggerPush().then(result => {
+        if (result === TriggerPushResult.Success) {
+          return this.markMetricsLastUploadedDateTime(new Date());
```

Comment:
Swap out the new Date here for 

getCurrentDate() - date-fns

Reply:
Will change